### PR TITLE
[Game engine] Game state

### PR DESCRIPTION
**Pull request by Rolf1e**
- [x] Remove `async`, `Arc`, `Mutex` from `Game`
- [x] Add observer logic for clients to hookup and get informed -> `GameStateObserver`
- [x] Players should be asked on their actions -> `TurnStrategy`
- [ ] Plug enum `Action` with Actix `Actors` to handle async actions on `Game`

### CODE REVIEW

File: rts-core/src/components/game.rs [hunk 1]
```diff
@@ -42,33 +60,37 @@ impl Game {
...
     }
 
-    /// Plays a turn with player execute given action
-    pub async fn play_with_async(&self, index: usize, action: Action) -> Result<(), RtsException> {
-        self.play_with(index, action)
+    fn play_with_all_players(&self) -> Result<(), RtsException> {
+        for (i, player) in self.players.iter().enumerate() {
+            let player_ptr = Rc::clone(player);
+            let action = player_ptr.borrow().request()?;
+            self.play(i, action)?;
```
Comment (cafehaine) [inline rts-core/src/components/game.rs:87]: If I understand this correctly, a player can only play one single action in a turn ?

File: rts-core/src/components/turn_strategy.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+use crate::entity::game_actions::Action;
+use crate::entity::player::TurnStrategyRequester;
+use crate::exceptions::RtsException;
+
+pub enum TurnStrategy {
+    AI,
```
Comment (cafehaine) [inline rts-core/src/components/turn_strategy.rs:6]: I guess this is here that we will add a variant that runs python code to get an action from a game state?

**Review by cafehaine — APPROVED** [review]
Everything looks good to me, well done!

File: rts-core/src/components/turn_strategy.rs (hunk 2 above)
Comment (Rolf1e) [reply rts-core/src/components/turn_strategy.rs:6]: Exactly !

File: rts-core/src/components/game.rs (hunk 1 above)
Comment (Rolf1e) [reply rts-core/src/components/game.rs:87]: Yup, maybe we will evolve this later. I guess this method should change with how we handle Actix actors

Comment (Rolf1e) [reply rts-core/src/components/game.rs:87]: Should we implement actors on this PR ?

Comment (cafehaine) [reply rts-core/src/components/game.rs:87]: I think it should be fine as is, we probably don't need actors just yet (or at all?)

**Pull request merged**